app/frontend/src/search/index.js

```diff
@@ -11,139 +9,118 @@ import { templates, renderContent } from './hits';
-            search(query);
-        },
-        onSubmit: (query) => {
-            if (shouldGeocode(query, locations)) {
-                getGeolocatedCoordinates(query).then(coords => {
-                    if (coords.success) {
-                        enableRadiusSelect();
-                        setDataAttribute(document.querySelector('#location'), 'coordinates', `${coords.lat}, ${coords.lng}`);
-                        setDataAttribute(document.querySelector('#radius'), 'radius', document.querySelector('#radius').value || 10);
-                        return searchClientInstance.refresh();
-                    }
-                });
-            }
+const locationSearchBox = searchBox({
+    container: document.querySelector('.filters-form'),
+    inputElement: document.getElementById('location'),
+    key: 'location',
+    autofocus: true,
+    queryHook(query, search) {
+        updateUrlQueryParams('location', document.querySelector('#location').value, window.location.href);
+        search(query);
+    },
+    onSubmit: query => onSubmit(query, locations, searchClientInstance)
+});
 
-            disableRadiusSelect();
-            removeDataAttribute(document.querySelector('#location'), 'coordinates');
-            removeDataAttribute(document.querySelector('#radius'), 'radius');
-            searchClientInstance.refresh();
-        }
-    });
+const keywordSearchBox = searchBox({
+    container: document.querySelector('.filters-form'),
+    inputElement: document.getElementById('keyword'),
+    key: 'keyword',
+    autofocus: true,
+    queryHook(query, search) {
+        updateUrlQueryParams('keyword', document.querySelector('#keyword').value, window.location.href);
+        search(query);
+    },
+    onSubmit: () => searchClientInstance.refresh()
+});
 
-    const keywordSearchBox = searchBox({
-        container: document.querySelector('.filters-form'),
-        inputElement: document.getElementById('keyword'),
-        key: 'keyword',
-        autofocus: true,
-        queryHook(query, search) {
-            updateUrlQueryParams('keyword', document.querySelector('#keyword').value, window.location.href);
-            search(query);
+searchClientInstance.addWidgets([
+    configure({
+        hitsPerPage: 10,
+    }),
+    autocomplete({
+        container: document.querySelector('#location-search'),
+        input: document.querySelector('#location'),
+        dataset: locations,
+        threshold: SEARCH_THRESHOLD,
+        onSelection: value => {
+            updateUrlQueryParams('location', value, window.location.href);
+        }
+    }),
+    locationSearchBox,
+    keywordSearchBox,
+    locationRadius({
+        container: document.querySelector('#location-radius-select'),
+        attribute: '_geoloc',
+        inputElement: document.getElementById('radius'),
+        onSelection: location => {
+            updateUrlQueryParams('radius', location, window.location.href);
```

Comment:
The location variable here seems misleading/confusing, could you use `radiusInMiles` instead?